### PR TITLE
Min Value Validation

### PR DESCRIPTION
**Pull request by hamishtaplin**
### What is the context of this PR?
Implements MinValue validation on numeric answers

### How to review 
- Run tests
- Ensure feature works as per https://trello.com/c/PfYRU46S/404-validation-min-value-front-end-changes

### Dependencies

API: https://github.com/ONSdigital/eq-author-api/pull/106
Schema: https://github.com/ONSdigital/eq-author-graphql-schema/pull/55

### CODE REVIEW

**Comment by adammahmood** [comment]
[Validation - Min value front end changes](https://trello.com/c/PfYRU46S/404-validation-min-value-front-end-changes)

**Comment by adammahmood** [comment]
@hamishtaplin Why have we co-located `withUpdateAnswerValidation` and `withToggleAnswerValidation` in the `Validation` folder? The current pattern is to locate them within `enhancers`

File: src/components/Validation/MinValue.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      enabled: value
+    };
+
+    onToggleValidationRule(toggleValidationRuleInput);
+  };
+
+  const handleMinValueChange = ({ value }) => {
+    // clamp value of input to +/- limit
+
+    if (value !== "" && !inRange(parseInt(value, 10), 0 - limit, limit + 1)) {
```
Comment (adammahmood) [inline src/components/Validation/MinValue.js]: @hamishtaplin In what circumstances does `value` equal an empty string?

File: cypress/integration/validation_spec.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          .type("3")
+          .should("have.value", "3");
+        cy.get("@minValueToggle").click(); //Turn off
+        cy.get("@minValueToggle").click(); //Turn on
+        cy.get("@minValueInput").should("have.value", "3");
+      });
+    });
+    afterEach(() => {
+      removeAnswer({ force: true });
+    });
```
[the remark this answers is not in the document]
Comment (adammahmood) [reply cypress/integration/validation_spec.js]: Tests don't necessarily have to be DRY.  I would imagine the different answer type tests to diverge over time.

File: cypress/integration/properties_spec.js [hunk 3]
```diff
@@ -193,12 +200,15 @@ describe("Answer Properties", () => {
...
-              .click({ force: true })
-              .should("not.be.checked");
+            cy.get(testId("answer-properties-toggle")).as("requiredToggle");
+            cy.get("@requiredToggle").click();
+            cy.get("@requiredToggle").within(() => {
+              cy.get('[type="checkbox"]').should("be.checked");
+            });
+            cy.get("@requiredToggle").click();
+            cy.get("@requiredToggle").within(() => {
+              cy.get('[type="checkbox"]').should("not.be.checked");
```
[the remark this answers is not in the document]
Comment (adammahmood) [reply cypress/integration/properties_spec.js]: Yes. Makes sense as we use this in the validation tests as well. +:1

File: package.json [hunk 4]
```diff
@@ -126,7 +126,6 @@
     "enzyme-to-json": "^3.3.1",
     "eq-author-graphql-schema": "0.21.0",
     "eslint-config-eq-author": "^1.1.0",
-    "extract-hoc": "^0.0.5",
```
Comment (adammahmood) [inline package.json]: @hamishtaplin I've removed this package as this PR[https://github.com/ONSdigital/eq-author/pull/403](url) should resolve any hot loading issues with HOC/React Context.

File: src/components/Validation/MinValue.js (hunk 1 above)
Comment (hamishtaplin) [reply src/components/Validation/MinValue.js]: when there are no characters in the text field (eg. when the value has been deleted)

File: package.json (hunk 4 above)
Comment (hamishtaplin) [reply package.json]: I thought I'd removed that, not supposed to be there

File: cypress/integration/properties_spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (adammahmood) [reply cypress/integration/properties_spec.js]: I've just looked into this further and I don't see the benefit of putting this into a util function as the 'properties_spec.js` is the only module that requires it.

**Pull request merged**